sprints/dashboard/utils.py

```diff
@@ -97,6 +100,37 @@ def get_cell_member_names(conn: CustomJira, members: Iterable[str]) -> Dict[str,
     """Returns cell members with their names."""
     return {conn.user(member).displayName: member for member in members}
 
+def get_cell_member_roles(raise_exception: bool = False) -> Dict[str, List[str]]:
+    """Return a dictionary of cell members and their associated roles.
+    Example return: {'John Doe': ['Recruitment Manager', 'Sprint Planning Manager',...],...}.
+
+    Note: We have room for error here if member names are spelled differently, hence we must
+    be careful that the handbook & Jira have consistent naming.
+    """
+
+    # Example HTML: <li><a href="../roles/#cell-manager-recruitment">Recruitment manager</a>: John Doe</li>
+    ROLES_REGEX = r"<li>.*roles.*>([A-Za-z ]+).*: (.+)<\/li>"
+
+    r = requests.get(HANDBOOK_ROLES_PAGE)
+
+    # roles = [('Recruitment manager', 'John Doe'),('Sprint Planning Manager', 'John Doe'),...]
+    roles = re.findall(ROLES_REGEX, r.text)
+
+    # roles_dict = {'John Doe': ['Recruitment Manager', 'Sprint Planning Manager',...],...}
+    roles_dict = {}
+    for role, member in roles:
+        if member not in roles_dict:
+            roles_dict[member] = [role]
+        else:
+            roles_dict[member].append(role)
+
+    # It's difficult to know whether we have read all roles
+    # So we assume that if we've read roles for less than 12 members, something's probably wrong
+    if raise_exception and len(roles_dict) < 12:
```

Comment:
This does not look right. Maybe let's just check if it's not empty instead of hardcoding this?
cc: @nizarmah

Reply:
> Maybe let's just check if it's not empty instead of hardcoding this?

Yeah I support just checking if it's empty instead of hard coding such a condition 👍🏼